crates/interpreter/src/side_table.rs

```diff
@@ -30,26 +47,43 @@ pub struct SideTableEntryView {
     pub pop_cnt: u32,
 }
 
-impl SideTableEntry {
+impl BranchTableEntry {
     const DELTA_IP_MASK: u64 = 0xffff;
-    const DELTA_STP_MASK: u64 = 0x7fff << 16;
-    const VAL_CNT_MASK: u64 = 0x3 << 31;
-    const POP_CNT_MASK: u64 = 0x3 << 33;
+    const DELTA_STP_MASK: u64 = 0xffff << 16;
+    const VAL_CNT_MASK: u64 = 0xf << 32;
+    const POP_CNT_MASK: u64 = 0xfff << 36;
 
-    pub fn new(view: SideTableEntryView) -> Result<Self, Error> {
+    pub fn new(view: BranchTableEntryView) -> Result<Self, Error> {
         let mut fields = 0;
         fields |= into_signed_field(Self::DELTA_IP_MASK, view.delta_ip)?;
         fields |= into_signed_field(Self::DELTA_STP_MASK, view.delta_stp)?;
         fields |= into_field(Self::VAL_CNT_MASK, view.val_cnt)?;
         fields |= into_field(Self::POP_CNT_MASK, view.pop_cnt)?;
-        Ok(SideTableEntry(fields))
+        Ok(BranchTableEntry(u64_to_u16_array(fields)))
     }
 
-    pub fn view(self) -> SideTableEntryView {
-        let delta_ip = from_signed_field(Self::DELTA_IP_MASK, self.0);
-        let delta_stp = from_signed_field(Self::DELTA_STP_MASK, self.0);
-        let val_cnt = from_field(Self::VAL_CNT_MASK, self.0);
-        let pop_cnt = from_field(Self::POP_CNT_MASK, self.0);
-        SideTableEntryView { delta_ip, delta_stp, val_cnt, pop_cnt }
+    pub fn view(self) -> BranchTableEntryView {
+        let entry = u16_array_to_u64(self.0);
+        let delta_ip = from_signed_field(Self::DELTA_IP_MASK, entry);
+        let delta_stp = from_signed_field(Self::DELTA_STP_MASK, entry);
+        let val_cnt = from_field(Self::VAL_CNT_MASK, entry);
+        let pop_cnt = from_field(Self::POP_CNT_MASK, entry);
+        BranchTableEntryView { delta_ip, delta_stp, val_cnt, pop_cnt }
     }
 }
+
+fn u64_to_u16_array(value: u64) -> [u16; 4] {
```

Comment:
We shouldn't implement those ourselves. There's 2 options: either use `bytemuck` or actually inline the bitfield logic. If we split like `(u16, u16, u4, u12)` then we don't really need to go through a `u64`. The `[u16; 4]` is close enough.